addons/remove-from-profile-rows/addon.json

```diff
@@ -0,0 +1,107 @@
+{
+  "name": "Button for removing items from profile rows",
+  "description": "On rows like \"Favorite Projects\" on your profile, add the option to remove projects/users from the list. For example, this will unfavorite a project when used on the \"Favorite Projects\" row.",
+  "tags": ["community", "profiles"],
+  "enabledByDefault": false,
+  "versionAdded": "1.32.0",
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["profiles", "https://scratch.mit.edu/users/*/*"]
+    }
+  ],
+  "userstyles": [
+    {
+      "url": "userstyle.css",
+      "matches": ["profiles", "https://scratch.mit.edu/users/*/*"]
+    }
+  ],
+  "settings": [
+    {
+      "name": "Only show on the dedicated pages",
+      "id": "dedicated_pages",
+      "type": "boolean",
+      "default": false
+    },
+    {
+      "name": "Only show the button on hover",
+      "id": "only_on_hover",
+      "type": "boolean",
+      "default": true
+    },
+    {
+      "name": "Show confirmation message before removing item",
+      "id": "show_confirmation",
+      "type": "boolean",
+      "default": true
+    },
+    {
+      "name": "Only show confirmation on destructive actions (Unsharing, leaving studios)",
+      "id": "only_destructive",
+      "type": "boolean",
+      "default": true,
+      "if": {
+        "settings": {"show_confirmation": true}
+      }
+    },
```

Comment:
I don't know about you, but I feel these two settings could be improved. The second setting here _removes_ functionality when _enabled_, and it's also slightly unclear what this should do if disabled - it seems to not imply by itself that when disabled, a confirmation will appear for all actions.

My idea to fix this problem would be to replace these on/off switches with a multi-choice setting named "confirmation before removing" with three choices, "off", "before unsharing projects and leaving studios" and "all actions", or something else.

Reply:
That does sound a lot better, and it also cuts down on the amount of settings required.

Reply:
I did that, but now it looks like this:
![image](https://user-images.githubusercontent.com/130385691/234474063-b9f9c256-b37b-46c2-977e-6653de8eb9d6.png)
And that isn't really good.

Reply:
The three-way setting could be first.

Reply:
![image](https://user-images.githubusercontent.com/130385691/234574372-388ff480-93fa-4db3-802c-ca61c5615f26.png)
That's better, but I don't think it's very ideal.

Reply:
I think when confirmations are disabled, it's expected that you won't get any confirmations.

Reply:
Done - https://github.com/ScratchAddons/ScratchAddons/pull/5960#issuecomment-1528868196